setup.py

```diff
@@ -142,17 +155,58 @@ def compile_cpp():
         target = os.path.abspath(os.path.join(os.path.pardir, 'bin', 'rgf'))
         logger.info("Trying to build executable file with g++ from existing makefile.")
         status = silent_call(('make'))
-        if status != 0 or not os.path.isfile(target) or not is_executable_response(target):
+        if not status or not os.path.isfile(target) or not is_executable_response(target):
             logger.warning("Building executable file with g++ "
                            "from existing makefile failed.")
             logger.info("Trying to build executable file with CMake.")
             clear_folder('.')
             status = silent_call(('cmake', '../'))
-            status += silent_call(('cmake', '--build', '.', '--config', 'Release'))
-        os.chdir(os.path.pardir)
-    os.chdir(os.path.join(os.path.pardir, os.path.pardir))
+            status &= silent_call(('cmake', '--build', '.', '--config', 'Release'))
+    os.chdir(CURRENT_DIR)
+    if status:
+        logger.info("Succeeded to build RGF.")
+    else:
+        logger.error("Compilation of rgf executable file failed. "
+                     "Please build from binaries by your own and "
+                     "specify path to the compiled file in the config file.")
+
+
+def compile_fastrgf():
+    def is_valid_gpp():
+        try:
+            result = subprocess.check_output(('g++', '--version'))
+            if sys.version > '3.0.0':
+                result = result.decode()
+            version = result.split('\n')[0].split(' ')[-2]
+            return version >= '4.8.0'
```

Comment:
I remember there were strange errors with big nthreads while compiled with g++ version less that 5.

Reply:
I do not know the reason, but if we call 'g++ --version' with travis, g++ 4.8.4 seems to be installed though we call 'install g++-5'.
https://travis-ci.org/fukatani/rgf_python/jobs/325070203 Please see l.1078

Reply:
Ah, I found two version g++ are exist in travis.
hmm...

Reply:
Yeah, g++ version 5 is called by `g++-5` command. OK, I'd better won't disturb you while you're in "deep WIP" 😄 .
Also please note that on macOS only g++-7 actually works due to hardcoded path in the CMakeLists.txt (I created a PR in fast_rgf repo yesterday to fix this).